### PR TITLE
User Navigation

### PR DESCRIPTION
**Pull request by hsmitha26**
Adds tests for registered user navigation.
Adds merchants_controller.
Adds Login form in sessions view.
Adds logout path and corresponding CRUD function in sessions controller.
CRUD function for login added in sessions controller.
All tests are passing.

### CODE REVIEW

**Comment by hsmitha26** [comment]
- Worked on getting all tests to pass.
- application.html.erb now has a new condition to control for existence of current user.

File: spec/features/users/navigation_spec.rb [hunk 1]
```diff
@@ -25,9 +25,17 @@
     end
   end
   describe 'As a registered user' do
-    xit 'I see the same links as a visitor' do
-      user = User.create(email: "j@gmail.com", password:"password", role:0)
-      visit root_path
+    it 'I see the same links as a visitor' do
```
Comment (jalena-penaligon) [inline spec/features/users/navigation_spec.rb]: I wonder if we can rephrase this slightly, since they see the same 1st 4 links, but then 2 links that are unique to a registered user. I think it may also be beneficial to add a test that we expect(page).to_not have_content for Login or Register?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect_to profile_path(user)
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    session[:user_id] = nil
```
[the remark this answers is not in the document]
Comment (hsmitha26) [reply app/controllers/sessions_controller.rb]: You are correct!  I will change the line to session.clear.  Thank you for catching that.

File: spec/features/users/navigation_spec.rb (hunk 1 above)
Comment (hsmitha26) [reply spec/features/users/navigation_spec.rb]: You are correct, again :)  I will change the wording to be more clear about what we are testing.

**Comment by hsmitha26** [comment]
Made the following changes:
- adds edge case tests for registered user navigation.
- refactored feature test for navigation.
- changed code to session.clear in sessions controller for logout action

**Comment by erin-king** [comment]
Looks good!
